[prawnzyy] iP

prawnzyy opened the pull request:
**Prawn**

> “How many programmers does it take to change a light bulb? None, that’s a hardware problem.” [(source)](http://www.jokeexplainer.com/2017/04/how-many-programmers-does-it-take-to.html)

Prawn frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is, 

1. download it from[ here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
___

If you Java programmer, you can use it to practice Java too. Here's the `main` method

```java
public class Main {
  public static void main(String[] args) {
    Application.launch(MainApp.class, args);
  }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

prawnzyy committed:
Completed Level 0

prawnzyy committed:
Completed Level 1

prawnzyy committed:
Completed Level 2

prawnzyy committed:
Completed Level 3

prawnzyy committed:
Bug fixes for Level-3

prawnzyy committed:
Completed Level 4

prawnzyy committed:
First Iteration of runtest

prawnzyy committed:
Completed Level-5 Tentatively

prawnzyy committed:
Second Iteration of runtest

prawnzyy committed:
Completed Level-6

prawnzyy committed:
Add Level-7

prawnzyy committed:
Add Level 7

prawnzyy committed:
Add Level-8

prawnzyy committed:
Add Level-8

prawnzyy committed:
Add A-MoreOOP

Many parts of code can be extracted out

Moved various parts of code into separate classes
to make code more user-friendly

prawnzyy committed:
Add A-MoreOOP

prawnzyy committed:
Move files to packages

Files too cluttered

Hard to view

Split into various packages for
easier viewing

prawnzyy committed:
Move files into packages

prawnzyy committed:
Merge gradle

prawnzyy committed:
Add gradle support

prawnzyy committed:
Add testing of Parser and TaskList

prawnzyy committed:
Create Jar File

prawnzyy committed:
Add Level 9

prawnzyy committed:
Add Javadocs

prawnzyy committed:
Update CodingStandard

prawnzyy committed:
Add Level 9

prawnzyy committed:
Add JavaDoc

prawnzyy committed:
Resolve merge conflict

prawnzyy committed:
Clean up code

prawnzyy committed:
Add config checkstyle

prawnzyy committed:
Add config checkstyle

prawnzyy committed:
Add Level-10

prawnzyy committed:
Add Level-10

Blizzeracz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+    private TaskList tasks;
+    private UI ui;
+    private String uiOutput;
+    private boolean justCreatedFile = false;
```
As much as possible, use a prefix such as is, has, was, etc. for boolean variable/method names so that linters can automatically verify that this style rule is being followed.

```suggestion
    private boolean hasCreatedFile = false;
```

Blizzeracz commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,108 @@
+     * @return A Command to be executed
+     * @throws DukeException If input does not match
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```
Maybe you could try incorporating the Switch/Case statement for this method if it is possible for better readability? 🤔

prawnzyy replied:
Ok sure, will try to make that change

Blizzeracz commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Just out if curiosity , why not make it private?

Blizzeracz reviewed (commented):
Overall, I found your code easy to read for the most part except that one place in parser
where the nesting was too deep. All of the classes are well designed and there is little to change. LGTM!!!

markgcera commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Override
+    public String fileRepresentation() {
```
Perhaps add a verb like "showFileRepresentation" to add a little bit more clarity?

prawnzyy committed:
Clean up code

prawnzyy committed:
Add Jar file

Original Jar file did not include javaFx

Rebuilt the jar file using the shadow plugin
to include javaFx

prawnzyy committed:
Merge branch 'master' of https://github.com/prawnzyy/ip

prawnzyy committed:
Add assert statements

Some portions of the code have assumptions
that it will work

Assertions have been added to check for such
assumptions and to terminate early if there
is any issue.

prawnzyy committed:
Made happy path more prominent

There are several methods where the happy path
is entangled within a few layers of if-else.

Moved the correct path out of the nesting and made
the logic for each path and guard clause clearer.

This should allow others to better understand the flow
of the program.

prawnzyy committed:
Merge pull request #1 from prawnzyy/branch-A-CodeQuality

Made happy path more prominent

prawnzyy committed:
Resolved merge conflicts

prawnzyy committed:
Merge pull request #2 from prawnzyy/branch-A-Assertions

Add assert statements

prawnzyy committed:
Add back accidentally removed assert statements

prawnzyy committed:
Add .yml file

prawnzyy committed:
Update .yml file

prawnzyy committed:
Fix checkstyle errors

prawnzyy committed:
Add sorting of tasks

Currently, tasks are sorted based on the order
they were inputted.

By sorting the task chronologically, it is easier to
see the tasks.

prawnzyy committed:
Merge branch 'branch-C-Sort'

prawnzyy committed:
Fix Checkstyle issues

prawnzyy committed:
Bug fixes

prawnzyy committed:
Update UI

prawnzyy committed:
Merge branch 'branch-A-BetterGui'

prawnzyy committed:
Add more Junit Test

prawnzyy committed:
Merge branch 'branch-A-MoreTesting'

prawnzyy committed:
Add readme

prawnzyy committed:
Update build.gradle
